snaps/openvino/gpu-support/snap/snapcraft.yaml

```diff
@@ -0,0 +1,145 @@
+    plugin: python
+    python-requirements: [src/bindings/python/requirements.txt]
+    python-packages: [ipython, jupyterlab, ipywidgets, opencv-python, tqdm, huggingface_hub, matplotlib>=3.4]
+    stage-packages:
+      - libva2 # openvino gpu plugin dep
+      - libgl1 # opencv-python dep
+      - libglu1-mesa # opencv-python dep
+      - libxcb1 # opencv-python dep
+      - libice6 # opencv-python dep
+      - libsm6 # opencv-python dep
+      - libx11-6 # opencv-python dep
+      - libxext6 # opencv-python-dep
+    stage:
+      - -usr/lib/x86_64-linux-gnu/libGLU.so.1.3.1
+      - -usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libicuio.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libicutest.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libvulkan.so.1.3.275
+      - -usr/local/lib/libopencl-clang.so
+      - -usr/lib/x86_64-linux-gnu/libOpenGL.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
+      - -usr/lib/x86_64-linux-gnu/libXxf86vm.so.1.0.0
+      - -usr/lib/x86_64-linux-gnu/libicutu.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-glx.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-present.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-randr.so.0.1.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-sync.so.1.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-xfixes.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxshmfence.so.1.0.0
+      - -usr/local/lib/libopencl-clang.so
+      - -usr/local/lib/libopencl-clang.so.14
+      - -usr/lib/x86_64-linux-gnu/libicui18n.so.74.2
+  opencl-driver:
+    # Includes all the compute runtime and OpenCL bits needed for Intel GPU support
+    plugin: nil
+    build-packages:
+      - wget
+    override-build: |
+      mkdir -p neo
+      cd neo
+      # Install Intel graphics compiler and compute runtime
+      wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.17791.9/intel-igc-core_1.0.17791.9_amd64.deb
+      wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.17791.9/intel-igc-opencl_1.0.17791.9_amd64.deb
+      wget https://github.com/intel/compute-runtime/releases/download/24.39.31294.12/intel-level-zero-gpu-dbgsym_1.6.31294.12_amd64.ddeb
+      wget https://github.com/intel/compute-runtime/releases/download/24.39.31294.12/intel-level-zero-gpu_1.6.31294.12_amd64.deb
+      wget https://github.com/intel/compute-runtime/releases/download/24.39.31294.12/intel-opencl-icd-dbgsym_24.39.31294.12_amd64.ddeb
+      wget https://github.com/intel/compute-runtime/releases/download/24.39.31294.12/intel-opencl-icd_24.39.31294.12_amd64.deb
+      wget https://github.com/intel/compute-runtime/releases/download/24.39.31294.12/libigdgmm12_22.5.2_amd64.deb
+      dpkg --root=$CRAFT_PART_INSTALL --force-all -i *.deb
+      # update paths to the Intel Installable Client Drivers (ICDs) for OpenCL
```

Comment:
i have a hard time understanding what we try to fix here about the ICD path, as the snap runs in a strict confinement context ... super grateful if you can help

Reply:
Sure, I'll add a comment for posterity. The issue is that the file uses a hard-coded path that points to a location on the host.

Reply:
but i way i understand snap is that for strict confinement, inside the snap, the rootfs is not the one from the host ... but let me check, do not spend more time debugging this, just a comment is enough for me

Reply:
That's not the case. Think about runtime environment variables like $LD_LIBRARY_PATH that need to prepend $SNAP. For example: https://github.com/canonical/tdx/blob/noble-24.04/tests/checkbox/bin/wrapper_local#L26-L31

Reply:
yes, but the checkbox provider has classic confinement : https://github.com/canonical/tdx/blob/198e71ae05ecdff5d751f3c4bbf9b19448ee4c95/tests/snap/snapcraft.yaml#L6C1-L7C1

Reply:
I do not say that inside a strict confined snap, you cannot see the `/snap/` folder contents, i m just saying that the `rootfs` (`/`) is not the one from host OS, that is my understanding, so when you said `...a hard-coded path that points to a location on the host.`, that confused me .... for example, if you do the listing of `/usr/bin` on the host and inside the snap, you probably will see different contents

Reply:
but let me think about it and take a look later when i will have some time, don't want to disturb you for this MR